### PR TITLE
feat: Add support for right side action on radio sections

### PR DESCRIPTION
**Pull request by gorandalum**
Necessary for the delete recipient flow.

Screenshot from design-system in app:
Screenshot 2024-08-09 at 11 12 40

### CODE REVIEW

File: src/components/sections/items/RadioSectionItem.tsx [hunk 1]
```diff
@@ -22,7 +23,13 @@ type Props = SectionItemProps<{
   mode?: ActionModes;
   accessibility?: AccessibilityProps;
   color?: InteractiveColor;
+  rightAction?: {
+    icon: (props: SvgProps) => JSX.Element;
+    onPress: () => void;
+    isLoading?: boolean;
+  };
```
Comment (marius-at-atb) [inline src/components/sections/items/RadioSectionItem.tsx:30]: Can this type be extracted into a variable and be reused both here and in RadioGroupSection?

**Review by marius-at-atb — APPROVED** [review]
Lgtm!

File: src/components/sections/items/RadioSectionItem.tsx (hunk 1 above)
Comment (marius-at-atb) [reply src/components/sections/items/RadioSectionItem.tsx:30]: Btw is it a bit strange that in one place rightAction is a function, and in another an object? Perhaps the names should be slightly different?

Comment (gorandalum) [reply src/components/sections/items/RadioSectionItem.tsx:30]: Good suggestion on the `itemToRightAction`, I'll do that!

**Pull request merged**